server/configs/passport-setup.js

```diff
@@ -0,0 +1,55 @@
+import passport from 'passport';
+import GoogleStrategy from 'passport-google-oauth20';
+import User from '../models/user';
+import { google } from './config';
+
+// After passport callback user will be passed here were it will serialize user.id
+// and pass it to cookie-session for encryption
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser(async (id, done) => {
+  let existingUser;
+  try {
+    if (id) {
+      existingUser = await User.findById(id);
+      if (existingUser.ok) {
+        done(null, existingUser);
+      } else {
+        done(null, null);
+      }
+    }
+    done(null, null);
+  } catch (e) {
+    console.log(e);
```

Comment:
always throw error instead of logging off, so that your app doesn't exist in inconsistent state